src/xbot/common/subsystems/BaseDriveSubsystem.java

```diff
@@ -0,0 +1,142 @@
+     */
+    public abstract double getTransverseDistance();
+    
+    /**
+     * Sets all the DriveSubsystem's "master" talons to a single control mode.
+     * e.g. PercentVBus, Follower, Current, etc...
+     */
+    protected void setTalonModes(TalonControlMode mode) {
+        updateLoggingLatch();
+        if (getAllMasterTalons() != null) {
+            getAllMasterTalons().keySet().stream()
+            .forEach((t) -> t.ensureTalonControlMode(mode));
+        }
+    }
+    
+    /**
+     * Commands each of the XCANTalons to respond to translation/rotation input.
+     * Each "wheel" is independently responsible, and as such there isn't any actual
+     * drive logic in this method.
+     * @param translation +Y is towards the front of the robot, +X is towards the right of the robot
+     * @param rotation +Rotation is left turn, -Rotation is right turn
+     * @param normalize If the largest output is greater than 1, should all outputs be
+     *        normalized so that 1 is the maximum?
+     */
+    public void drive(XYPair translation, double rotation, boolean normalize) {
+        updateLoggingLatch();
+        Map<XCANTalon, MotionRegistration> talons = getAllMasterTalons();
+        
+        if (talons != null) {
+            double normalizationFactor = 1;
+            if (normalize) {
+                normalizationFactor = Math.max(1, getMaxOutput(translation, rotation));
+            }            
+            
+            for(Map.Entry<XCANTalon, MotionRegistration> entry : talons.entrySet()) {
+               entry.getKey().ensureTalonControlMode(TalonControlMode.PercentVbus);
+               double power = entry.getValue()
+                       .calculateTotalImpact(translation.x, translation.y, rotation) / normalizationFactor;
+               entry.getKey().set(power);
+            }
+        }
+    }
+    
+    /**
+     * Commands each of the XCANTalons to respond to translation/rotation input.
+     * Each "wheel" is independently responsible, and as such there isn't any actual
+     * drive logic in this method.
+     * @param translation +Y is towards the front of the robot, +X is towards the right of the robot
+     * @param rotation +Rotation is left turn, -Rotation is right turn
+     */
+    public void drive(XYPair translation, double rotation) {
+        drive(translation, rotation, false);
+    }
+    
+    public void stop() {
+        drive(new XYPair(0, 0), 0);
+    }
+    
+    /**
+     * Determine the largest commanded output for a given wheel for a given
+     * translation/rotation input. For example, you would see a maximum output of
+     * two in a Tank Drive system that's commanded to go foward at full speed AND
+     * rotate at full speed.
+     */
+    protected double getMaxOutput(XYPair translation, double rotation) {
+        return getAllMasterTalons().values().stream()
+        .map(mr -> mr.calculateTotalImpact(translation.x, translation.y, rotation))
```

Comment:
Give `mr` a more readable name -- perhaps `master`

Reply:
It took me a minute to remember what MR was supposed to stand for, which is not a good sign. =]

"motion registration"

Fixing.